Canvas/GradesWidget/GradesTodayWidgetViewController.swift

```diff
@@ -277,19 +163,62 @@ extension GradesTodayWidgetViewController: UITableViewDataSource {
         cell.textLabel!.numberOfLines = 0
         return cell
     }
+
+    func colorForCourse(course: Core.Course) -> UIColor? {
+        guard let color = presenter.colors.filter({ $0.canvasContextID == course.canvasContextID }).first else {
+            return nil
+        }
+        return color.color
+    }
+
+    func displayGrade(for course: Core.Course) -> String {
+        guard let enrollments = course.enrollments, let enrollment = enrollments.filter({ $0.role == .student }).first else {
+            return ""
+        }
+        
+        let grade = enrollment.multipleGradingPeriodsEnabled
+            ? enrollment.currentPeriodComputedCurrentGrade
+            : enrollment.computedCurrentGrade
+
+        let formatter = NumberFormatter()
```

Comment:
It's usually best practice not to init new formatters in `cellForRowAtIndexPath`.   Have you considered making this formatter a class var to help table view methods be more efficient?